pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -382,3 +399,48 @@ func (c *DeploymentConfigController) handleErr(err error, key interface{}) {
 		c.queue.Forget(key)
 	}
 }
+
+// cleanupOldDeployments triggers on deployment configuration update and add events in order
+// to delete old replication controller deployments if their quota has been reached
+func (c *DeploymentConfigController) cleanupOldDeployments(existingDeployments []kapi.ReplicationController, deploymentConfig *deployapi.DeploymentConfig) error {
+	if deploymentConfig.Spec.RevisionHistoryLimit == nil {
+		// there is no past deplyoment quota set
+		return nil
+	}
+
+	activeDeployment := deployutil.ActiveDeployment(deploymentConfig, existingDeployments)
+
+	// we need to coerce the list type into something we can sort, and we need to filter out
+	// any deployments that we don't care about, namely the active deployment and any newer
+	// deployments
+	oldDeployments := []*kapi.ReplicationController{}
+	for i := 0; i < len(existingDeployments); i++ {
+		if &existingDeployments[i] == activeDeployment || deployutil.DeploymentVersionFor(&existingDeployments[i]) > deployutil.DeploymentVersionFor(activeDeployment) {
+			continue
+		}
+		oldDeployments = append(oldDeployments, &existingDeployments[i])
+	}
+
+	if len(oldDeployments) <= *deploymentConfig.Spec.RevisionHistoryLimit {
+		// the past deployment quota has not been exceeded
+		return nil
+	}
+
+	sort.Sort(controller.ControllersByCreationTimestamp(oldDeployments))
```

Comment:
I don't remember off-hand. Does this sort from old to new or vice versa? Can you add a comment clarifying that we need to start deleting older deployments first?